client/components/payment-activity/date-range-picker.tsx

```diff
@@ -0,0 +1,155 @@
+
+const now = moment();
+const yesterdayEndOfDay = moment()
+	.clone()
+	.subtract( 1, 'd' )
+	.set( { hour: 23, minute: 59, second: 59, millisecond: 0 } );
+const todayEndOfDay = moment()
+	.clone()
+	.set( { hour: 23, minute: 59, second: 59, millisecond: 0 } );
+const timeOptions: {
+	[ key: string ]: {
+		start: moment.Moment;
+		end: moment.Moment;
+		displayKey: string;
+	};
+} = {
+	today: {
+		start: now
+			.clone()
+			.set( { hour: 0, minute: 0, second: 0, millisecond: 0 } ),
+		end: todayEndOfDay,
+		displayKey: __( 'Today', 'woocommerce-payments' ),
+	},
+	last_7_days: {
+		start: now
+			.clone()
+			.subtract( 7, 'days' )
+			.set( { hour: 0, minute: 0, second: 0, millisecond: 0 } ),
+		end: yesterdayEndOfDay,
+		displayKey: __( 'Last 7 days', 'woocommerce-payments' ),
+	},
+	last_4_weeks: {
+		start: now
+			.clone()
+			.subtract( 4, 'weeks' )
+			.set( { hour: 0, minute: 0, second: 0, millisecond: 0 } ),
+		end: yesterdayEndOfDay,
+		displayKey: __( 'Last 4 weeks', 'woocommerce-payments' ),
+	},
+	last_3_months: {
+		start: now
+			.clone()
+			.subtract( 3, 'months' )
+			.set( { hour: 0, minute: 0, second: 0, millisecond: 0 } ),
+		end: yesterdayEndOfDay,
+		displayKey: __( 'Last 3 months', 'woocommerce-payments' ),
+	},
+	last_12_months: {
+		start: now
+			.clone()
+			.subtract( 12, 'months' )
+			.set( { hour: 0, minute: 0, second: 0, millisecond: 0 } ),
+		end: yesterdayEndOfDay,
+		displayKey: __( 'Last 12 months', 'woocommerce-payments' ),
+	},
+	month_to_date: {
+		start: now.clone().startOf( 'month' ),
+		end: todayEndOfDay,
+		displayKey: __( 'Month to date', 'woocommerce-payments' ),
+	},
+	quarter_to_date: {
+		start: now.clone().startOf( 'quarter' ),
+		end: todayEndOfDay,
+		displayKey: __( 'Quarter to date', 'woocommerce-payments' ),
+	},
+	year_to_date: {
+		start: now.clone().startOf( 'year' ),
+		end: todayEndOfDay,
+		displayKey: __( 'Year to date', 'woocommerce-payments' ),
+	},
+};
+
+const formatDateRange = (
+	start: moment.Moment,
+	end: moment.Moment
+): string => {
+	// Today - show only today's date.
+	if ( start.isSame( end, 'day' ) ) {
+		return start.format( 'MMMM D, YYYY' );
+	}
+
+	// Different years - show year for both start and end
+	if ( ! start.isSame( end, 'year' ) ) {
+		return `${ start.format( 'MMMM D, YYYY' ) } - ${ end.format(
+			'MMMM D, YYYY'
+		) }`;
+	}
+
+	// Same year - show year only for end date.
+	return `${ start.format( 'MMMM D' ) } - ${ end.format( 'MMMM D, YYYY' ) }`;
+};
+
+const options = Object.keys( timeOptions ).map( ( optionKey ) => {
+	const option = timeOptions[ optionKey ];
+	return {
+		key: optionKey,
+		name: option.displayKey,
+		hint: formatDateRange( option.start, option.end ),
+	};
+} );
+
+export const defaultDateRange = {
+	key: 'last_7_days',
+	date_start: timeOptions.last_7_days.start.format( 'YYYY-MM-DD\\THH:mm:ss' ),
+	date_end: timeOptions.last_7_days.end.format( 'YYYY-MM-DD\\THH:mm:ss' ),
+};
+
+export const DateRangePicker: React.FC< {
```

Comment:
(minor) Is this a generally useful date range picker, or can we use a more specific name? 

It looks to me like this component is designed for reporting/analytics, so personally I'd call it something like:

- `AnalyticsTimePeriodSelect`
- `ReportingTimePeriodSelect`
- `StatisticsTimePeroodSelect`
- `TimePeriodSelect`

`Picker` might make more sense for future (adding in a date picker), but we aren't there yet (and can rename later!).

Could also focus on the current use-case - selecting a preset. 

Could also research other products/libraries to see what they name this component (e.g. Google Analytics, Facebook/Meta ads/marketing dashboard etc). Or ask design what they call it. @rogermattic ?

Note I have no objections to `DateRangePicker`, works for me.

Reply:
This component / file has been removed via [c373c4f](https://github.com/Automattic/woocommerce-payments/pull/8927/commits/c373c4f86c250fd3e65d3d522f94b64297be4192) . Marking as resolved

Reply:
This is a valid point. The component file has been deleted while we simplified the dropdown within the index file. Hence, marking this as resolved. [c373c4f](https://github.com/Automattic/woocommerce-payments/pull/8927/commits/c373c4f86c250fd3e65d3d522f94b64297be4192)